cocos/2d/components/ui-opacity.ts

```diff
@@ -125,14 +127,22 @@ export class UIOpacity extends Component {
             return;
         }
 
-        const render = node._uiProps.uiComp as UIRenderer;
         const uiOp = node.getComponent(UIOpacity);
-
         if (uiOp && stopRecursiveIfHasOpacity) {
             // Because it's possible that UiOpacity components are handled by themselves (at onEnable or onDisable)
             uiOp._parentOpacity = parentOpacity;
             return;
         }
+
+        // If the node has never been loaded, then node._uiProps.uiComp won't be set.
+        // It has to be obtained by getComponent at this time.
+        let render = null as UIRenderer | null;
+        if (node._objFlags & IsOnLoadCalled) {
```

Comment:
Is it possible to use the _isOnLoadCalled interface directly?

Reply:
Probably not. The `_isOnLoadCalled` interface belongs to `Component`.
I can't guarantee `uiOp` or `render` will be there at this point.

Reply:
ok，got it.

Reply:
[Fix the UIRenderer check logic in UIOpacity.](https://github.com/cocos/cocos-engine/pull/17508)
@qiuguohua
@minggo